Disfunction/Assets/Game Scripts/Development/GadgetScripts/DroneScripts/DroneMechanics.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DroneMechanics : MonoBehaviour
+{
+
+    public float moveSpeed;
+    public float upSpeed;
+    public float downSpeed;
+    public float rotationSpeed;
+    public float cameraSensitivity;
+
+    private bool moveForward;
+    private bool moveBackward;
+    private bool moveUp;
+    private bool moveDown;
+    private bool moveLeft;
+    private bool moveRight;
+    private bool rotateRight;
+    private bool rotateLeft;
+
+    public Transform cameraTransform;
+
+    private void Start()
+    {
+        
+    }
+
+    private void Update()
+    {
+        checkClick();
+        move();
+        cameraMovement();
+    }
+
+    private void cameraMovement()
+    {
+        float mouseY = Input.GetAxis("Mouse Y");
+
+        float verticalRotation = cameraTransform.localEulerAngles.x - mouseY * cameraSensitivity * Time.deltaTime;
+
+        // Adjust for Unity's 0-360 degree rotation system
+        if (verticalRotation > 180) verticalRotation -= 360;
+
+        // Clamp the rotation between -25 and 25 degrees
+        verticalRotation = Mathf.Clamp(verticalRotation, -25f, 35f);
+
+        // Apply the clamped rotation
+        cameraTransform.localRotation = Quaternion.Euler(verticalRotation, cameraTransform.localEulerAngles.y, cameraTransform.localEulerAngles.z);
+
+
+    }
+
+    private void checkClick()
```

Comment:
Put all transform functionalities into switch cases since we don't need to use them simultaneously.

Reply:
couldn't use switch case in this scenario